cli/azd/pkg/azsdk/zip_deploy_client.go

```diff
@@ -0,0 +1,208 @@
+	StatusText   string     `json:"status_text"`
+	Message      string     `json:"message"`
+	Progress     *string    `json:"progress"`
+	ReceivedTime *time.Time `json:"received_time"`
+	StartTime    *time.Time `json:"start_time"`
+	EndTime      *time.Time `json:"end_time"`
+	Complete     bool       `json:"complete"`
+	Active       bool       `json:"active"`
+	LogUrl       string     `json:"log_url"`
+	SiteName     string     `json:"site_name"`
+}
+
+func NewZipDeployClient(
+	subscriptionId string,
+	credential azcore.TokenCredential,
+	options *arm.ClientOptions,
+) (*ZipDeployClient, error) {
+	if options == nil {
+		options = &arm.ClientOptions{}
+	}
+
+	// We do not have a Resource provider to register
+	options.DisableRPRegistration = true
+
+	pipeline, err := armruntime.NewPipeline("zip-deploy", "1.0.0", credential, runtime.PipelineOptions{}, options)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating HTTP pipeline: %w", err)
+	}
+
+	return &ZipDeployClient{
+		subscriptionId: subscriptionId,
+		pipeline:       pipeline,
+	}, nil
+}
+
+// Begins a zip deployment and returns a poller to check for status
+func (c *ZipDeployClient) BeginDeploy(
+	ctx context.Context,
+	appName string,
+	zipFile io.Reader,
+) (*runtime.Poller[*DeployResponse], error) {
+	request, err := c.createDeployRequest(ctx, appName, zipFile)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := c.pipeline.Do(request)
+	if err != nil {
+		return nil, runtime.NewResponseError(response)
+	}
+
+	defer response.Body.Close()
+
+	if !runtime.HasStatusCode(response, http.StatusAccepted) {
+		return nil, runtime.NewResponseError(response)
+	}
+
+	var finalResponse *DeployResponse
+
+	pollerOptions := &runtime.NewPollerOptions[*DeployResponse]{
+		Response: &finalResponse,
+		Handler:  newDeployPollingHandler(c.pipeline, response),
+	}
+
+	return runtime.NewPoller(response, c.pipeline, pollerOptions)
+}
+
+// Deploys the specified application zip to the azure app service and waits for completion
+func (c *ZipDeployClient) Deploy(ctx context.Context, appName string, zipFile io.Reader) (*DeployResponse, error) {
+	poller, err := c.BeginDeploy(ctx, appName, zipFile)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := poller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{
+		Frequency: deployStatusInterval,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// Creates the HTTP request for the zip deployment operation
+func (c *ZipDeployClient) createDeployRequest(
+	ctx context.Context,
+	appName string,
+	zipFile io.Reader,
+) (*policy.Request, error) {
+	endpoint := fmt.Sprintf("https://%s.scm.azurewebsites.net/api/zipdeploy", appName)
+	req, err := runtime.NewRequest(ctx, http.MethodPost, endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("creating deploy request: %w", err)
+	}
+
+	rawRequest := req.Raw()
+	rawRequest.Body = io.NopCloser(zipFile)
+	query := rawRequest.URL.Query()
+	query.Set("isAsync", "true")
```

Comment:
Can we include a link somewhere in this file to the REST API specification? I suspect it might be this: https://github.com/MicrosoftDocs/azure-docs/blob/main/includes/app-service-deploy-zip-push-rest.md